src/Components/Game.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import Board from "./Board";
+import { GameProps, isNotNull } from "../Types/TypeProps";
+import GameInfo from './GameInfo';
+/**
+ * Handle Game logic
+ * @param props 
+ * @returns 
+ */
+const Game = (props:GameProps) => {
+  /**
+   * Game's state declarations
+   */
+  const [history, setHistory] = useState(props.historyVal);
```

Comment:
I would explicitly type this

```suggestion
  const [history, setHistory] = useState<{ squares: Array<null | string> }[]>(props.historyVal);
```

You'll have to update this type for all your functions too. I would just put it into it's own type like this:

```ts
type GameHistory = { squares: Array<null | string> }[]
```

If you put this type in a type definition file (.d.ts), you'll get access to the type everywhere without exporting/importing.

```suggestion
  const [history, setHistory] = useState<GameHistory>(props.historyVal);
```